Core/src/Geometry/BlueprintNode.cpp

```diff
@@ -0,0 +1,170 @@
+// This file is part of the ACTS project.
+//
+// Copyright (C) 2016 CERN for the benefit of the ACTS project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "Acts/Geometry/BlueprintNode.hpp"
+
+#include "Acts/Geometry/CylinderContainerBlueprintNode.hpp"
+#include "Acts/Geometry/LayerBlueprintNode.hpp"
+#include "Acts/Geometry/MaterialDesignatorBlueprintNode.hpp"
+#include "Acts/Geometry/StaticBlueprintNode.hpp"
+#include "Acts/Navigation/INavigationPolicy.hpp"
+#include "Acts/Navigation/TryAllNavigationPolicy.hpp"
+
+#include <ostream>
+
+namespace Acts {
+
+namespace {
+bool hasDescendent(const BlueprintNode& descendent,
+                   const BlueprintNode& ancestor) {
+  if (&descendent == &ancestor) {
+    return true;
+  }
+
+  for (const auto& child : ancestor.children()) {
+    if (hasDescendent(descendent, child)) {
+      return true;
+    }
+  }
```

Comment:
could potentially be done with a range function?

Reply:
This requires https://github.com/acts-project/acts/pull/3971, which for now is included in this PR.